pkg/controller/migplan/migplan_controller.go

```diff
@@ -360,9 +408,118 @@ func (r *ReconcileMigPlan) planSuspended(plan *migapi.MigPlan) error {
 	return nil
 }
 
+func (r *ReconcileMigPlan) AddUpdateMigrationStateToPlan(plan *migapi.MigPlan, state string) error {
+
+	migrationStateInstances := 0
+	needsUpdate := false
+	migrations, err := plan.ListMigrations(r)
+	if err != nil {
+		return liberr.Wrap(err)
+	}
+
+	//Check if there are any associated migrations with the migplan for the given state (additional condition for failed state)
+	if state == migctl.Failed {
+		for _, m := range migrations {
+			if m.Status.HasCondition(state) || m.Status.HasCondition(migctl.PlanNotReady) {
+				migrationStateInstances++
+			}
+		}
+	} else {
+		for _, m := range migrations {
+			if m.Status.HasCondition(state) {
+				migrationStateInstances++
+			}
+		}
+	}
+
+	planLabel := getLabelForPlan(state)
+
+	switch {
+
+	case migrationStateInstances == 0 && plan.Labels != nil:
+		//There migrations associated with the Migplan for the given state
+		if _, found := plan.Labels[planLabel]; found {
+			delete(plan.Labels, planLabel)
+			needsUpdate = true
+		}
+
+	case migrationStateInstances > 0:
+		//Migplan is associated with migrations for the given state
+		if plan.Labels == nil {
+			plan.Labels = make(map[string]string)
+		}
+		plan.Labels[planLabel] = "true"
+		needsUpdate = true
+	}
+
+	if !needsUpdate {
+		return nil
+	}
+
+	return liberr.Wrap(r.Update(context.TODO(), plan))
+}
+
+func getLabelForPlan(state string) string {
+
+	planLabel := ""
+
+	switch {
+
+	case state == migctl.Running:
+		planLabel = migapi.MigplanMigrationRunning
+
+	case state == migctl.Failed:
+		planLabel = migapi.MigplanMigrationFailed
+	}
+
+	return planLabel
+}
+
 // Update Status.ExcludedResources based on settings
 func (r *ReconcileMigPlan) setExcludedResourceList(plan *migapi.MigPlan) error {
 	excludedResources := Settings.Plan.ExcludedResources
 	plan.Status.ExcludedResources = excludedResources
 	return nil
 }
+
+func (r ReconcileMigPlan) deleteImageRegistryResourcesForClient(client k8sclient.Client, plan *migapi.MigPlan) error {
```

Comment:
@djwhatle I think keeping the `ForClient` will help answer questions like why is the `r.Client` not being used here. The `ForClient` explicitly tells me that a different client than the controller cluster client is being passed for a reason that there could be source and destination client

Reply:
That makes sense to me, it just differs from the norms in the controller. I think there's some value in having a consistent naming convention for functions that do similar things for codebase readability. If we use this here perhaps we should also consider matching the convention elsewhere.

Reply:
I'll also add that it's very normal for us to be using a passed in client in the migration controller since connecting to remote clusters is very commonplace here. 

We would really need to attach `forClient` to a lot of our functions with this convention.